File: library/src/main/java/com/shxhzhxx/sdk/ui/CheckBox.java
package com.shxhzhxx.sdk.ui;

import android.content.Context;
import android.content.res.ColorStateList;
import android.content.res.TypedArray;
import android.graphics.Color;
import androidx.core.widget.CompoundButtonCompat;
import android.util.AttributeSet;
import android.util.TypedValue;

import com.shxhzhxx.sdk.R;

public class CheckBox extends androidx.appcompat.widget.AppCompatCheckBox {
    public CheckBox(Context context) {
        this(context, null);
    }

    public CheckBox(Context context, AttributeSet attrs) {
        this(context, attrs, androidx.appcompat.R.attr.checkboxStyle);
    }

    public CheckBox(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        TypedValue typedValue = new TypedValue();
        getContext().getTheme().resolveAttribute(R.attr.colorPrimary, typedValue, true);
        int primaryColor = typedValue.data;

        int color_for_state_checked, color_for_state_normal;
        TypedArray a = context.getTheme().obtainStyledAttributes(attrs,
                R.styleable.CheckBox, 0, 0);
        try {
            color_for_state_checked = a.getColor(R.styleable.CheckBox_checkBoxCheckedColor, primaryColor);
            color_for_state_normal = a.getColor(R.styleable.CheckBox_checkBoxColor, Color.GRAY);
        } finally {
            a.recycle();
        }
        int states[][] = {{android.R.attr.state_checked}, {}};
        int colors[] = {color_for_state_checked, color_for_state_normal};
        CompoundButtonCompat.setButtonTintList(this, new ColorStateList(states, colors));
    }
}

File: library/src/main/java/com/shxhzhxx/sdk/ui/CheckBox.kt
package com.shxhzhxx.sdk.ui

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.Color
import android.util.AttributeSet
import android.util.TypedValue
import androidx.appcompat.widget.AppCompatCheckBox
import androidx.core.widget.CompoundButtonCompat
import com.shxhzhxx.sdk.R

class CheckBox @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null,
                                         defStyleAttr: Int = R.attr.checkboxStyle) : AppCompatCheckBox(context, attrs, defStyleAttr) {
    init {
        val primaryColor = TypedValue().also { getContext().theme.resolveAttribute(R.attr.colorPrimary, it, true) }.data
        val array = context.theme.obtainStyledAttributes(attrs, R.styleable.CheckBox, 0, 0)
        val (colorChecked, colorNormal) = try {
            array.getColor(R.styleable.CheckBox_checkBoxCheckedColor, primaryColor) to array.getColor(R.styleable.CheckBox_checkBoxColor, Color.GRAY)
        } catch (ignore: UnsupportedOperationException) {
            primaryColor to Color.GRAY
        } finally {
            array.recycle()
        }
        val states = arrayOf(intArrayOf(android.R.attr.state_checked), intArrayOf())
        val colors = intArrayOf(colorChecked, colorNormal)
        CompoundButtonCompat.setButtonTintList(this, ColorStateList(states, colors))
    }
}
